bread/layout/components/tooltip.py

```diff
@@ -0,0 +1,268 @@
+            It can be either start, center, or end.
+            The default value is 'center'.
+        position : str
+            Where should the tooltip appear besides the tooltip icon.
+            It can be either top, left, right, or bottom.
+            The default value is 'bottom'.
+        """
+        tooltip_attributes = {
+            "_class": (
+                "bx--tooltip__trigger "
+                "bx--tooltip--a11y "
+                "bx--tooltip--%s "
+                "bx--tooltip--align-%s"
+            )
+            % (position, align),
+            "data_tooltip_icon": True,
+        }
+
+        tooltip_attributes = hg.merge_html_attrs(tooltip_attributes, attributes)
+        icon = _get_icon(icon)
+
+        super().__init__(
+            hg.BUTTON(
+                hg.SPAN(description, _class="bx--assistive-text"),
+                icon,
+                **tooltip_attributes,
+            )
+        )
+
+
+class InteractiveTooltip(hg.BaseElement):
+    """Interactive tooltip should be used if there are actions a user can take in
+    the tooltip (e.g. a link or a button). For more regular use case, e.g. giving
+    the user more text information about something, use definition tooltip or icon
+    tooltip.
+
+    Reference: https://the-carbon-components.netlify.app/?nav=tooltip
+    """
+
+    base_class = "bx--tooltip"
+    suffix_label = "label"
+
+    def __init__(
+        self,
+        label: Union[hg.BaseElement, str],
+        body: str,
+        heading: Optional[str] = None,
+        link: Optional[Link] = None,
+        button: Optional[Button] = None,
+        icon: Union[Icon, str] = "information",
+        menudirection: str = "bottom",
+        **attributes,
+    ):
+        """
+        Parameters
+        ----------
+        label : str
+            Label for a tooltip
+        body : str
+            The content inside a tooltip
+        heading : str, optional
+            A heading for a tooltip.
+        link : Link, optional
+            Bread's Link NamedTuple in case you want to bring users to a specific webpage.
+        button : Button, optional
+            Insert the Bread's Button onto the tooltip.
+        icon : Icon, str, optional
+            Specify an icon for the tooltip
+            The default value is "information".
+        menudirection : str
+            Where should the tooltip appear besides the tooltip icon.
+            It can be either top, left, right, or bottom.
+            The default value is "bottom".
+        """
+        footer = (
+            hg.BaseElement(
+                hg.If(bool(link), hg.A(link.label, href=link.href, _class="bx--link")),
+                hg.If(bool(button), button),
+            )
+            if link or button
+            else None
+        )
+
+        base_id = hg.html_id(self, self.base_class + "-id")
+        label_id = "-".join([base_id, self.suffix_label])
```

Comment:
I would use an f-string instead of a join here, since you are only joining two strings

Reply:
Thank you for the suggestion.